File: DesignPatternsExercises/java/FacadeDesignPattern/Exercise/BookAvailability.java
import java.util.HashMap;
import java.util.Map;

public class BookAvailability {
	private static Map<String, Integer> bookMap ;
	static {
		bookMap = new HashMap<>();
		bookMap.put(Constants.JAVA_BOOK, 1);
		bookMap.put(Constants.DESIGN_BOOK, 3);
		bookMap.put(Constants.MISTAKES_BOOK, 7);
	}

	public boolean __________(1)__________(String book) {
		if(bookMap.containsKey(book) && bookMap.get(book) > __________(2)__________) {
			System.out.println(book + " is available");
			return true;
		} else {
			System.out.println("Sorry, " + book + " is not available");
			return false;
		}
	}

	public void bookCheckout(String book) {
		bookMap.put(book, bookMap.get(book) - 1);
		System.out.println("You have 2 days to return it");
	}
}

File: DesignPatternsExercises/ruby/facade_design_pattern/exercise/book_availability.rb
class BookAvailability

  def __________(2)__________.create_book_map
    __________(3)__________ = { Constants::JAVA_BOOK => 1, Constants::DESIGN_BOOK => 3, Constants::MISTAKES_BOOK => 7 }
  end

  def book_available?(book)
    if (@@book_map.include? book) && (@@book_map[book] > 0)
      puts "#{book} is available"
      true
    else
      puts "Sorry, #{book} is not available"
      false
    end
  end

  def book_checkout(book)
    new_book_map = { book => @@book_map[book] - 1 }
    @@book_map.update(new_book_map)
    puts 'You have 2 days to return it'
  end
end
